frontend/app/helpers/spree/taxons_helper.rb

```diff
@@ -4,12 +4,13 @@ module TaxonsHelper
     # that we can use configurations as well as make it easier for end users to override this determination.  One idea is
     # to show the most popular products for a particular taxon (that is an exercise left to the developer.)
     def taxon_preview(taxon, max=4)
-      products = taxon.active_products.select("DISTINCT (spree_products.id), spree_products.*, spree_products_taxons.position").limit(max)
+      products = taxon.active_products.select("spree_products.*, spree_products_taxons.position").limit(max)
       if (products.size < max)
         products_arel = Spree::Product.arel_table
         taxon.descendants.each do |taxon|
           to_get = max - products.length
-          products += taxon.active_products.select("DISTINCT (spree_products.id), spree_products.*, spree_products_taxons.position").where(products_arel[:id].not_in(products.map(&:id))).limit(to_get)
+          products += taxon.active_products.select("spree_products.*, spree_products_taxons.position")
+                      .where(products_arel[:id].not_in(products.map(&:id))).limit(to_get)
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Align `.where` with `.active_products` on line 12.